docs/advanced.rst

```diff
@@ -1018,105 +1136,665 @@ to learn more about the use of this feature.
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-SERVER-LOCATORS<-->      |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+Note that a *server* can act as a client of other *servers*, thus, the following section may also apply.
+
+Client side setup
+""""""""""""""""""
+
+Each *client* must keep a list of locators associated to the *servers* it wants to link to.
+Each *server* specifies its own locators.
+The locator list is the attribute:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.builtin.discovery_config.m_DiscoveryServers
+
+and must be populated with ``RemoteServerAttributes`` objects with a valid ``metatrafficUnicastLocatorList`` or
+``metatrafficMulticastLocatorList`` member. In xml the server list is and its elements are simultaneously specified.
+Note the ``metatrafficUnicastLocatorList`` or ``metatrafficMulticastLocatorList`` attributes of the ``RemoteServer``
+tag.
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>builtin>discovery_config>discoveryServerList>RemoteServer@metatrafficUnicastLocatorList
+    dds>profiles>participant>rtps>builtin>discovery_config>discoveryServerList>RemoteServer@metatrafficMulticastLocatorList
+
++------------------------------------------------------------+
+| **C++**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp                 |
+|    :language: c++                                          |
+|    :start-after: //CONF_SERVER_CLIENT_LOCATORS             |
+|    :end-before: //!--                                      |
++------------------------------------------------------------+
+| **XML**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-CLIENT-LOCATORS<-->      |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+.. _DS_ping_period:
+
+Client ping period
+^^^^^^^^^^^^^^^^^^
+
+As explained :ref:`above <DS_key_concepts>` the *clients* send hailing messages to the *servers* at regular intervals
+(ping period) until they received message reception acknowledgement.
+This period is specified in the member:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.builtin.discovery_config.discoveryServer_client_syncperiod
+
+or the xml tag:
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>builtin>discovery_config>clientAnnouncementPeriod
```

Comment:
Get rid of the enum and tag hierarchy, as they are implicit on the examples.

Reply:
As above... It's not easy to imply the jigsaw puzzle that the attributes are, directly, from the examples...that's the main reason I put these clues there.

Reply:
👍

Reply:
Is the xml tag correct? It does not match with the attribute

Reply:
Yes, I wasn't careful enough to change it when the parsing was implemented.